es6/experiment.js

```diff
@@ -100,7 +100,20 @@ class Experiment {
   }
 
   assign(params, args) {
-    throw "IMPLEMENT THIS";
+    throw "IMPLEMENT assign";
+  }
+
+  experimentParameters() {
```

Comment:
Perhaps this could instead be called `getParamNames()` to make it clear that the function returns an array of parameter names?  In general it would be helpful if all of the methods had some lightweight annotations to denote their return types, even if you aren't using a type checker like Flow or TypeScript.